pkg/loop/internal/chain_reader.go

```diff
@@ -0,0 +1,60 @@
+package internal
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/loop/internal/pb"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+var _ types.ChainReader = (*chainReaderClient)(nil)
+
+type chainReaderClient struct {
+	*brokerExt
+	grpc pb.ChainReaderClient
+}
+
+func (c *chainReaderClient) GetLatestValue(ctx context.Context, bc types.BoundContract, method string, params, retVal any) error {
+	boundContract := pb.BoundContract{Name: bc.Name, Address: bc.Address, Pending: bc.Pending}
+	jsonParams, err := json.Marshal(params)
+	if err != nil {
+		return err
+	}
+
+	reply, err := c.grpc.GetLatestValue(ctx, &pb.GetLatestValueRequest{Bc: &boundContract, Method: method, Params: jsonParams})
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(reply.RetVal, retVal)
+}
+
+var _ pb.ChainReaderServer = (*chainReaderServer)(nil)
+
+type chainReaderServer struct {
+	pb.UnimplementedChainReaderServer
+	impl types.ChainReader
+}
+
+func (c *chainReaderServer) GetLatestValue(ctx context.Context, request *pb.GetLatestValueRequest) (*pb.GetLatestValueReply, error) {
+	var bc types.BoundContract
+	bc.Name = request.Bc.Name[:]
+	bc.Address = request.Bc.Address[:]
+	bc.Pending = request.Bc.Pending
+	var retVal []byte
+	err := c.impl.GetLatestValue(ctx, bc, request.Method, request.Params, retVal)
+	if err != nil {
+		return nil, err
+	}
+	return &pb.GetLatestValueReply{RetVal: retVal}, nil
```

Comment:
Don't we need to `json.Unmarshal` the `Params` (to a map) and `json.Marshsal` the retval (from a map)?
There should be symmetry - I expected something like:
```go
func (r chainReaderClient) GetLatestValue(bc BoundContract, method string, params, returnVal any) error {
	req, err := json.Marshal(params)
	if err != nil {
		return err
	}
	resp, err := r.client.GetLatestValue(bc, method, req)
	if err != nil {
		return err
	}
	return json.Unmarshal(resp, returnVal)
}

func (r chainReaderServer) GetLatestValue(bc BoundContract, method string, params) ([]byte, error) {
	var req map[string]any{}
	err := json.Unmarshal(params, &req)
	if err != nil {
		return err
	}
	resp, err := r.srv.GetLatestValue(bc, method, req)
	if err != nil {
		return err
	}
	return json.Marshal(resp)
}
``

Reply:
```suggestion
func (c *chainReaderServer) GetLatestValue(ctx context.Context, request *pb.GetLatestValueRequest) (*pb.GetLatestValueReply, error) {
	var params map[string]any{}
	err := json.Unmarshal(request.Params, &params) 
	if err != nil {
		return err
	}
	var bc types.BoundContract
	bc.Name = request.Bc.Name[:]
	bc.Address = request.Bc.Address[:]
	bc.Pending = request.Bc.Pending
	var retVal map[string]any{}
	err := c.impl.GetLatestValue(ctx, bc, request.Method, params, &retVal)
	if err != nil {
		return nil, err
	}
	b, err := json.Marshal(retVal)
	return &pb.GetLatestValueReply{RetVal: b}, err
```